public/controllers/agent/components/register-agent.js

```diff
@@ -206,7 +206,7 @@ export class RegisterAgent extends Component {
         {}
       );
       const remote = ((result.data || {}).data || {}).remote || {};
-      return (remote[0] || {}).protocol !== 'tcp';
+      return (remote[0] || {}).protocol !== 'tcp' && (remote[0] || {}).protocol[0] !== 'TCP';
```

Comment:
it could  be improved with lower case and checking is an array
```
const protocol = Array.isArray((remote[0] || {}).protocol) ?  remote[0].protocol[0] : (remote[0] || {}).protocol || '';
return protocol.toUpperCase() !== 'TCP';
```